packages/carousel/src/react/index.tsx

```diff
@@ -0,0 +1,341 @@
+  pages: () => compose(css(stylesheet['.psds-carousel__pages'])),
+  page: (isActivePage: boolean) =>
+    compose(
+      css(stylesheet['.psds-carousel__page']),
+      isActivePage && css(stylesheet['.psds-carousel__page--active'])
+    ),
+  instructions: () => css(stylesheet['.psds-carousel__instructions']),
+  item: () => css(stylesheet['.psds-carousel__item'])
+}
+
+interface CarouselProps extends React.HTMLAttributes<HTMLDivElement> {
+  children: React.ReactNode
+  controlPrev?: React.ReactNode
+  controlNext?: React.ReactNode
+  size?: ValueOf<typeof vars.sizes>
+}
+interface CarouselStatics {
+  Control: typeof Control
+  Item: typeof Item
+  sizes: typeof vars.sizes
+}
+type CarouselComponent = React.FC<CarouselProps> & CarouselStatics
+
+const Carousel: CarouselComponent = ({
+  children,
+  controlPrev,
+  controlNext,
+  size,
+  ...rest
+}) => {
+  const id = useUniqueId('carousel-')
+  const ref = React.useRef<HTMLDivElement>(null)
+  const { width } = useResizeObserver(ref)
+
+  controlPrev = controlPrev || <Control direction={Control.directions.prev} />
+  controlNext = controlNext || <Control direction={Control.directions.next} />
+  size = size || Carousel.sizes.narrow
+
+  const constraints = vars.constraints[size]
+  const perPage = calcItemsPerPage(constraints, width)
+
+  const childArr = React.Children.toArray(children) as React.ReactElement[]
+  const pages = chunk<React.ReactElement>(childArr, perPage).map(page =>
+    fillPageToMaintainItemWidth<React.ReactElement>(page, perPage)
+  )
+  const pageCount = pages.length
+
+  const pager = usePager(pageCount, [width])
+  const ready = !!width && width > 0
+  const [transitioning, setTransitioning] = React.useState<boolean>(false)
+  const contextValue = {
+    ...pick(pager, ['activePage', 'next', 'offset', 'prev']),
+    id,
+    pageCount,
+    perPage,
+    transitioning,
+    setTransitioning
+  }
+
+  return (
+    <CarouselContext.Provider value={contextValue}>
+      <div {...styles.carousel(ready)} {...rest} ref={ref}>
+        {controlPrev}
+        <Pages
+          ref={pager.ref}
+          id={id}
+          onKeyDown={evt => {
+            if (isLeftArrow(evt)) pager.prev()
+            if (isRightArrow(evt)) pager.next()
+          }}
+          onSwipeLeft={pager.next}
+          onSwipeRight={pager.prev}
+        >
+          {pages.map((items, pageIndex) => {
+            const isActivePage = pager.activePage === pageIndex
+
+            return (
+              <Page
+                key={pageIndex}
+                isActivePage={isActivePage}
+                paged={pager.paged}
+              >
+                {items.map(
+                  (item: React.ReactElement | null, itemIndex: number) => {
+                    const wrapped = !!item && isOfComponentType(item, Item)
+                    if (!wrapped) item = <Item key={itemIndex}>{item}</Item>
+
+                    // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
```

Comment:
one way to do this is to `{React.isActiveElement(item) && ...}`

Reply:
seems like I tried that option.  Not sure what the outcome was or why I chose this one.  There's a weekend in between in my mind.  

Is there an approach you think we should prefer?